app/components/contents/external_file_component.rb

```diff
@@ -4,15 +4,10 @@ module Contents
   # Displays external files for a virtual object
   # e.g. https://argo.stanford.edu/view/druid:tm280sk2404
   class ExternalFileComponent < ViewComponent::Base
-    def initialize(external_file:)
-      @resource_id, @filename = external_file.split('/')
+    def initialize(druid)
```

Comment:
the convention for view components is to use kwargs (see https://viewcomponent.org/). You'll also need to update the caller to provide this kwarg rather than external_file:

Reply:
Ok, thanks for the huddle -- I just pushed up the `with_collection_parameter` change.